pkg/status/status.go

```diff
@@ -345,91 +324,13 @@ func getCommonStatus() (map[string]interface{}, error) {
-	json.Unmarshal(autoConfigStatsJSON, &autoConfigStats) //nolint:errcheck
-	stats["autoConfigStats"] = autoConfigStats
-
-	checkSchedulerStatsJSON := []byte(expvar.Get("CheckScheduler").String())
-	checkSchedulerStats := make(map[string]interface{})
-	json.Unmarshal(checkSchedulerStatsJSON, &checkSchedulerStats) //nolint:errcheck
-	stats["checkSchedulerStats"] = checkSchedulerStats
-
-	aggregatorStatsJSON := []byte(expvar.Get("aggregator").String())
-	aggregatorStats := make(map[string]interface{})
-	json.Unmarshal(aggregatorStatsJSON, &aggregatorStats) //nolint:errcheck
-	stats["aggregatorStats"] = aggregatorStats
-	s, err := checkstats.TranslateEventPlatformEventTypes(stats["aggregatorStats"])
-	if err != nil {
-		log.Debugf("failed to translate event platform event types in aggregatorStats: %s", err.Error())
-	} else {
-		stats["aggregatorStats"] = s
-	}
-
-	if expvar.Get("dogstatsd") != nil {
-		dogstatsdStatsJSON := []byte(expvar.Get("dogstatsd").String())
-		dogstatsdUdsStatsJSON := []byte(expvar.Get("dogstatsd-uds").String())
-		dogstatsdUDPStatsJSON := []byte(expvar.Get("dogstatsd-udp").String())
-		dogstatsdStats := make(map[string]interface{})
-		json.Unmarshal(dogstatsdStatsJSON, &dogstatsdStats) //nolint:errcheck
-		dogstatsdUdsStats := make(map[string]interface{})
-		json.Unmarshal(dogstatsdUdsStatsJSON, &dogstatsdUdsStats) //nolint:errcheck
-		for name, value := range dogstatsdUdsStats {
-			dogstatsdStats["Uds"+name] = value
-		}
-		dogstatsdUDPStats := make(map[string]interface{})
-		json.Unmarshal(dogstatsdUDPStatsJSON, &dogstatsdUDPStats) //nolint:errcheck
-		for name, value := range dogstatsdUDPStats {
-			dogstatsdStats["Udp"+name] = value
-		}
-		stats["dogstatsdStats"] = dogstatsdStats
-	}
-
-	pyLoaderData := expvar.Get("pyLoader")
-	if pyLoaderData != nil {
-		pyLoaderStatsJSON := []byte(pyLoaderData.String())
-		pyLoaderStats := make(map[string]interface{})
-		json.Unmarshal(pyLoaderStatsJSON, &pyLoaderStats) //nolint:errcheck
-		stats["pyLoaderStats"] = pyLoaderStats
-	} else {
-		stats["pyLoaderStats"] = nil
-	}
-
-	pythonInitData := expvar.Get("pythonInit")
-	if pythonInitData != nil {
-		pythonInitJSON := []byte(pythonInitData.String())
-		pythonInit := make(map[string]interface{})
-		json.Unmarshal(pythonInitJSON, &pythonInit) //nolint:errcheck
-		stats["pythonInit"] = pythonInit
-	} else {
-		stats["pythonInit"] = nil
-	}
-
-	hostnameStatsJSON := []byte(expvar.Get("hostname").String())
-	hostnameStats := make(map[string]interface{})
-	json.Unmarshal(hostnameStatsJSON, &hostnameStats) //nolint:errcheck
-	stats["hostnameStats"] = hostnameStats
-
-	ntpOffset := expvar.Get("ntpOffset")
-	if ntpOffset != nil && ntpOffset.String() != "" {
-		stats["ntpOffset"], err = strconv.ParseFloat(expvar.Get("ntpOffset").String(), 64)
-	}
+func expvarStats(stats map[string]interface{}) (map[string]interface{}, []error) {
+	stats, errors := expvarcollector.Report(stats)
 
 	inventories := expvar.Get("inventories")
```

Comment:
I couldn't find where the `"inventories"` get created. Is this variable still in use?